cloudsync/tests/test_cs.py

```diff
@@ -819,6 +819,36 @@ def test_cs_basic(cs):
     assert not cs.state.changeset_len
 
 
+def test_cs_stop(four_local_cs):
+    syncs = list(four_local_cs)
+    for cs in syncs:
+        cs.start()
+
+    # stop a single cloudsync
+    syncs[0].stop()
+    for et in syncs[0].ethreads:
+        assert not et.is_alive()
+
+    # stop(join=False) + join()
+    with patch.object(syncs[1], "sthread") as sthread:
+        syncs[1]._stop(join=False)
+        assert syncs[1].sthread
+        sthread.join.assert_not_called()
+
+        syncs[1]._join()
+        assert not syncs[1].sthread
+        sthread.join.assert_called_once()
+
+    # make sure the Thread object patched above is still a Thread, so that it can be joined below
+    assert isinstance(syncs[1].sthread, threading.Thread)
```

Comment:
This is a race condition. The fact that this is a Thread now does not mean it will still be one a millisecond from now

Reply:
removed